dlc/src/test/scala/org/bitcoins/dlc/BinaryOutcomeDLCWithSelfTest.scala

```diff
@@ -9,23 +9,92 @@ import org.bitcoins.core.crypto.{
+}
 import org.bitcoins.core.script.crypto.HashType
 import org.bitcoins.core.util.{BitcoinScriptUtil, CryptoUtil}
-import org.bitcoins.core.wallet.fee.SatoshisPerByte
-import org.bitcoins.core.wallet.utxo.P2PKHSpendingInfo
+import org.bitcoins.core.wallet.builder.{BitcoinTxBuilder, TxBuilder}
+import org.bitcoins.core.wallet.fee.{FeeUnit, SatoshisPerByte}
+import org.bitcoins.core.wallet.utxo.{
+  BitcoinUTXOSpendingInfo,
+  P2PKHSpendingInfo
+}
+import org.bitcoins.testkit.core.gen.{
+  CurrencyUnitGenerator,
+  TransactionGenerators
+}
 import org.bitcoins.testkit.util.BitcoinSAsyncTest
+import org.scalacheck.Gen
 import org.scalatest.Assertion
 import scodec.bits.ByteVector
 
-import scala.concurrent.Future
+import scala.concurrent.{Await, Future}
 import scala.concurrent.duration.DurationInt
 
 class BinaryOutcomeDLCWithSelfTest extends BitcoinSAsyncTest {
   behavior of "BinaryOutcomeDLCWithSelf"
 
+  it should "correctly subtract fees evenly amongst outputs" in {
+    // Can't use TransactionGenerators.realisiticOutputs as that can return List.empty
+    val nonEmptyRealisticOutputsGen = Gen
+      .choose(1, 5)
+      .flatMap(n => Gen.listOfN(n, TransactionGenerators.realisticOutput))
+      .suchThat(_.nonEmpty)
+
+    // CurrencyUnitGenerator.feeRate gives too high of fees
+    val feeRateGen = Gen.choose(0, CurrencyUnits.oneBTC.satoshis.toLong).map {
+      n =>
+        SatoshisPerByte(Satoshis(Int64(n)))
+    }
+
+    forAll(nonEmptyRealisticOutputsGen, feeRateGen) {
+      case (outputs, feeRate) =>
+        val totalInput = outputs.foldLeft(CurrencyUnits.zero) {
+          case (accum, output) =>
+            accum + output.value
+        }
+
+        val inputKey = ECPrivateKey.freshPrivateKey
+        val utxos: Vector[BitcoinUTXOSpendingInfo] = Vector(
+          P2PKHSpendingInfo(TransactionOutPoint(DoubleSha256DigestBE.empty,
+                                                UInt32.zero),
+                            totalInput,
+                            P2PKHScriptPubKey(inputKey.publicKey),
+                            inputKey,
+                            HashType.sigHashAll))
+        val changeSPK = EmptyScriptPubKey
+        val network: BitcoinNetwork = RegTest
+
+        val txBuilderF =
+          BitcoinTxBuilder(outputs, utxos, feeRate, changeSPK, network)
+
+        val badFeeF = txBuilderF.flatMap { txBuilder =>
+          recoverToSucceededIf[IllegalArgumentException](txBuilder.sign)
+        }
+
+        val resultF = for {
+          txBuilder <- txBuilderF
+          _ <- badFeeF
+          tx <- BinaryOutcomeDLCWithSelf.subtractFeeAndSign(txBuilder)
+        } yield {
+          val diffs = outputs.zip(tx.outputs).map {
+            case (before, after) =>
+              before.value - after.value
+          }
+
+          val firstDiff = diffs.head
+          // Fee has been evenly distributed (up to some remainder)
+          assert(diffs.forall(diff => diff.compare(firstDiff) < diffs.length))
```

Comment:
I don't understand what this is actually checking? `diff.compare(firstDiff)` just checks returns `-1`, `0`, or `1` doesn't it?